src/main/java/de/tum/in/www1/artemis/service/ProgrammingAssessmentService.java

```diff
@@ -33,4 +40,53 @@ public Result updateAssessmentAfterComplaint(Result originalResult, Exercise exe
         originalResult.setScore(assessmentUpdate.getScore());
         return super.updateAssessmentAfterComplaint(originalResult, exercise, assessmentUpdate);
     }
+
+    /**
+     * This function is used for saving a manual assessment/result. It sets the assessment type to MANUAL and sets the assessor attribute. Furthermore, it saves the result in the
+     * database.
+     *
+     * @param result the new result of a programming exercise
+     * @return result that was saved in the database
+     */
+    @Transactional
+    public Result saveManualAssessment(Result result) {
+        Boolean isFeedbackEmpty = result.getFeedbacks().isEmpty();
+        result.setHasFeedback(!isFeedbackEmpty);
+
+        User user = userService.getUserWithGroupsAndAuthorities();
+        result.setHasComplaint(false);
+        result.setAssessmentType(AssessmentType.MANUAL);
+        result.setAssessor(user);
+
+        result.setCompletionDate(null);
+        result.setRated(false);
+
+        result.getFeedbacks().forEach(feedback -> {
+            feedback.setResult(result);
+        });
+
+        // Note: This also saves the feedback objects in the database because of the 'cascade = CascadeType.ALL' option.
+        return resultRepository.save(result);
+    }
+
+    /**
+     * This function is used for submitting a manual assessment/result. It gets the result that belongs to the given resultId, updates the completion date, sets the assessment type
+     * to MANUAL and sets the assessor attribute. Afterwards, it saves the update result in the database again.
+     *
+     * @param resultId the id of the result that should be submitted
+     * @return the ResponseEntity with result as body
+     */
+    @Transactional
```

Comment:
I am not that familiar with hibernate, but why are you using @Transactional here?

Reply:
To be honst, I am neither. I orientated myself to how it is done for the other exercise types. There, we used `@Transactional` see https://github.com/ls1intum/Artemis/blob/develop/src/main/java/de/tum/in/www1/artemis/service/ModelingAssessmentService.java#L79
that's why I used it as well to keep it consistent to the other exercises.